pkg/api/latest/latest.go

```diff
@@ -99,8 +101,34 @@ func InterfacesFor(version string) (*meta.VersionInterfaces, error) {
 }
 
 func init() {
+	// The list of valid API versions to test that KUBE_API_VERSIONS does not contain an invalid version.
+	// Note that the list is in ascending sorted order.
+	validAPIVersions := []string{"v1", "v1beta1", "v1beta2", "v1beta3"}
+	// Env var KUBE_API_VERSIONS is a comma separated list of supported API versions.
+	// The versions should be in the order of most preferred to the least.
+	supportedVersions := os.Getenv("KUBE_API_VERSIONS")
+	if supportedVersions == "" {
+		supportedVersions = "v1beta3,v1beta1,v1beta2,v1"
+	}
+	versions := strings.Split(supportedVersions, ",")
+	// The first version in the list is the latest version.
+	Version = versions[0]
+	Codec = runtime.CodecFor(api.Scheme, Version)
+	// Put the versions in Versions in reverse order.
+	Versions = []string{}
+	for i := len(versions) - 1; i >= 0; i-- {
+		version := versions[i]
+		// Verify that the version is valid.
+		searchIndex := sort.SearchStrings(validAPIVersions, version)
+		if searchIndex == len(validAPIVersions) || validAPIVersions[searchIndex] != version {
+			// Not a valid API version.
+			panic(fmt.Sprintf("invalid api version: %s in KUBE_API_VERSIONS: %s. List of valid API versions: %s", version, os.Getenv("KUBE_API_VERSIONS"), strings.Join(validAPIVersions, ", ")))
```

Comment:
Discussed this in person. We dont require panic here, since the stack trace is not required.
Updated the code to use glog.Fatalf